// sensei/dist/core/messages/core/excecoes-messages.js
export const ExcecoesMensagens = {
    exit1: 'exit:1',
    requireMutateFsAutoFix: 'Auto-fix indisponível',
    autoFixTimeout: (timeoutMs) => `Auto-fix timeout após ${timeoutMs}ms`,
    pluginsDesabilitadosSafeMode: 'Carregamento de plugins desabilitado em SAFE_MODE. Defina SENSEI_ALLOW_PLUGINS=1 para permitir.',
    pluginBloqueado: (erro) => `Plugin bloqueado: ${erro}`,
    caminhoPluginNaoResolvido: 'Caminho de plugin não resolvido',
    pluginRegistradoNaoPodeSerObtido: (name) => `Plugin ${name} está registrado mas não pode ser obtido`,
    pluginCarregandoPromiseNaoPodeSerObtida: (name) => `Plugin ${name} está sendo carregado mas promise não pode ser obtida`,
    naoFoiPossivelCarregarPlugin: (name, errMsg) => `Não foi possível carregar o plugin '${name}': ${errMsg}`,
    pluginDeveTerNomeValido: 'Plugin deve ter um nome válido',
    pluginDeveTerVersaoValida: 'Plugin deve ter uma versão válida',
    pluginDeveDefinirPeloMenosUmaExtensao: 'Plugin deve definir pelo menos uma extensão',
    pluginDeveImplementarMetodoParse: 'Plugin deve implementar método parse()',
    definicaoAnalistaInvalida: 'Definição de analista inválida',
    analistaSemFuncaoAplicar: (nome) => `Analista ${nome} sem função aplicar`,
    caminhoForaDaCwdNaoPermitido: (p) => `Caminho fora da CWD não permitido: ${p}`,
    persistenciaNegadaForaRaizProjeto: (caminho) => `Persistência negada: caminho fora da raiz do projeto: ${caminho}`,
    fsWriteFileBinaryIndisponivel: 'fs.writeFile (binary) indisponível no ambiente atual',
    fsReadFileIndisponivel: 'fs.readFile indisponível no ambiente atual',
    fsWriteFileIndisponivel: 'fs.writeFile indisponível no ambiente atual',
    fsRenameIndisponivel: 'fs.rename indisponível no ambiente atual',
    fsMkdirIndisponivel: 'fs.mkdir indisponível no ambiente atual',
    versaoSchemaDesconhecida: (versao) => `Versão de schema desconhecida: ${versao}`,
    relatorioSchemaInvalido: (erros) => `Relatório com schema inválido: ${erros}`,
    arquivoNaoEncontrado: (fileCaminho) => `Arquivo não encontrado: ${fileCaminho}`,
    validacaoFalhouPara: (fileCaminho) => `Validação falhou para ${fileCaminho}`,
    erroAoLer: (fileCaminho, errMsg) => `Erro ao ler ${fileCaminho}: ${errMsg}`,
    erroAoEscrever: (fileCaminho, errMsg) => `Erro ao escrever ${fileCaminho}: ${errMsg}`,
    erroAoDeletar: (fileCaminho, errMsg) => `Erro ao deletar ${fileCaminho}: ${errMsg}`,
    statIndefinidoPara: (fullCaminho) => `Stat indefinido para ${fullCaminho}`,
    mapaReversaoCorrompido: 'Mapa de reversão corrompido',
    semPkg: 'sem pkg'
};
