// sensei/dist/core/messages/cli/cli-processamento-diagnostico-messages.js
import { ICONES_ACAO, ICONES_ARQUIVO, ICONES_FEEDBACK, ICONES_STATUS } from '../ui/icons.js';
export const CliProcessamentoDiagnosticoMensagens = {
    linhaEmBranco: '',
    templateArquetipoPreview: 'Template de arquétipo personalizado gerado (pré-visualização)',
    resumoExecutivoHeader: (total, criticos, altos) => `Resumo Executivo: ${total} problemas (críticos: ${criticos}, altos: ${altos})`,
    resumoExecutivoCriticos: (criticos) => `Críticos: ${criticos}`,
    resumoExecutivoAltos: (altos) => `Altos: ${altos}`,
    resumoExecutivoTopArquivosErrosAltos: 'Top arquivos (erros/altos):',
    resumoExecutivoBulletTopArquivo: (arquivo, count) => `  • ${arquivo} (${count})`,
    resumoExecutivoAcaoSugerida: 'Ação sugerida: filtrar por arquivo com --include "<padrão>" ou usar --export para análise off-line',
    principaisTiposTitulo: '  Mensagens  Principais tipos:',
    principaisTiposLinha: (tipo, count) => `     • ${tipo}: ${count}`,
    topArquivosTitulo: `  ${ICONES_ARQUIVO.diretorio} Top arquivos:`,
    topArquivosLinha: (arquivo, count) => `     • ${arquivo} (${count})`,
    porTipoTitulo: 'Por tipo:',
    porTipoLinha: (tipo, count) => `  • ${tipo}: ${count}`,
    porSeveridadeTitulo: 'Por severidade:',
    porSeveridadeLinha: (prefixo, nivel, count) => `  ${prefixo} ${nivel}: ${count}`,
    arquivosMaisOcorrenciasTitulo: 'Arquivos com mais ocorrências',
    topArquivosPorOcorrenciasTitulo: 'Top arquivos por ocorrências:',
    topCriticosTitulo: 'Top críticos (10)',
    topCriticosTituloComDoisPontos: 'Top críticos (10):',
    topCriticosLinha: (nivelUpper, tipo, arquivo, pos) => `  • ${nivelUpper} | ${tipo} — ${arquivo}${pos}`,
    amostraOcorrenciasTitulo: 'Amostra de ocorrências:',
    amostraLinhaIndentada: (linha) => `  ${linha}`,
    amostraMaisLinhas: (shown, total) => `  ... mostrando ${shown}/${total}. Use --export ou ajustar filtros para estudar todas.`,
    amostraBlocoTitulo: (count) => `Amostra (${count})`,
    analistaOcorrenciasLinha: (nome, ocorrencias, duracaoMsFixed) => `  • ${nome}: ${ocorrencias} ocorrências em ${duracaoMsFixed}ms`,
    diagnosticoConcluido: 'Diagnóstico concluído',
    tudoPronto: 'Tudo pronto',
    arquetiposDetectados: (qtd) => `Detectados ${qtd} candidatos a arquétipos`,
    arquetiposCompact: (nome, confidence) => `Arquétipos: ${nome} (${confidence}%)`,
    arquetiposCandidatosEncontrados: (qtd) => `Arquétipos candidatos encontrados: ${qtd}`,
    arquetiposCandidatosTitulo: 'Arquétipos candidatos:',
    arquetiposCandidatoLinha: (nome, confidence) => `  ${nome} (${confidence}%)`,
    planoSugestaoMove: (qtd) => `planoSugestao: ${qtd} move`,
    planoSugestaoNenhumMove: 'planoSugestao: nenhum move sugerido',
    conflitos: (qtd) => `conflitos: ${qtd}`,
    anomaliasTitulo: 'Anomalias detectadas',
    anomaliasTituloComDoisPontos: 'Anomalias detectadas:',
    anomaliaLinha: (path, motivo) => `${path}: ${motivo}`,
    anomaliasMais: (qtdExtra) => `... e mais ${qtdExtra} anomalias`,
    anomaliasOcultasAviso: (qtdOcultas) => `Há ${qtdOcultas} anomalias ocultas. Use --full para ver todas.`,
    driftAlterou: (anterior, atual) => `drift: arquétipo alterou de ${anterior} para ${atual}`,
    driftMantido: (atual) => `drift: arquétipo ${atual} mantido`,
    driftNovos: (novosStr) => `novos:[${novosStr}]`,
    driftRemovidos: (removidosStr) => `removidos:[${removidosStr}]`,
    baselineDesconhecidoAviso: 'Baseline desconhecido - primeira execução ou baseline não encontrado',
    baselineArquetipoDesconhecido: 'Arquétipo baseline: desconhecido',
    baselineArquetipo: (arquetipo, confidence) => `Arquétipo baseline: ${arquetipo} (${confidence}%)`,
    baselineCriadoEm: (ptBr) => `Baseline criado em: ${ptBr}`,
    driftDetectado: (anterior, atual) => `Drift detectado: ${anterior} → ${atual}`,
    arquetipoMantido: (atual) => `Arquétipo mantido: ${atual}`,
    novosArquivosRaiz: (joined) => `Novos arquivos na raiz: ${joined}`,
    arquivosRemovidosRaiz: (joined) => `Arquivos removidos da raiz: ${joined}`,
    candidatoPrincipal: (nome, confidence) => `Candidato principal: ${nome} (${confidence}%)`,
    resumoEstruturaTitulo: 'Resumo da estrutura',
    debugAboutToEmitJson: (optsJson) => `About to emit JSON output; opts=${optsJson}`,
    todosPendentesEncontrados: (qtd) => `${qtd} TODOs pendentes encontrados`,
    errorGeneratingJson: 'Error generating JSON:',
    fallbackJson: 'Fallback JSON:',
    resumoExecutivoTitulo: (prioritarios) => `Resumo Executivo (${prioritarios} prioritários)`,
    cabecalhoExecProblema: 'Problema',
    cabecalhoExecQtd: 'Qtd',
    dicaUseFull: (total) => `${ICONES_FEEDBACK.dica} Use --full para ver todos os ${total} problemas encontrados`,
    projetoBomEstado: (total) => `${ICONES_STATUS.ok} Projeto em bom estado! Apenas ${total} problemas menores encontrados.`,
    quickFixesDisponiveis: (qtd) => `${ICONES_ACAO.correcao} ${qtd} correções automáticas disponíveis`,
    resumoTiposTitulo: 'Resumo dos tipos de problemas',
    cabecalhoResumoTipo: 'Tipo',
    cabecalhoResumoQuantidade: 'Quantidade',
    relatorioFullFragmentado: (manifestFile) => `Relatório full fragmentado. Manifest: ${manifestFile}`,
    relatorioScanSalvo: (dir) => `Relatório de scan salvo em ${dir}`,
    falhaExportarRelatorioScanOnly: (errMsg) => `Falha ao exportar relatório de scan-only: ${errMsg}`,
    autoFixModoAgressivo: 'Modo agressivo ativado - aplicando mais correções (use com cuidado)',
    falhaGerarRelatorioMarkdownMetadados: (errMsg) => `Falha ao gerar relatório Markdown com metadados: ${errMsg}`,
    relatoriosExportadosPara: (dir) => `Relatórios exportados para ${dir}`,
    falhaSalvarRelatorioJson: (errMsg) => `Falha ao salvar relatório JSON: ${errMsg}`,
    falhaExportarRelatorios: (errMsg) => `Falha ao exportar relatórios: ${errMsg}`,
    erroFatalDiagnostico: (errMsg) => `Erro fatal durante o diagnóstico: ${errMsg}`
};
