// sensei/dist/core/messages/cli/cli-comando-reestruturar-messages.js
import { ICONES_ACAO, ICONES_COMANDO, ICONES_DIAGNOSTICO, ICONES_STATUS } from '../ui/icons.js';
export const CliComandoReestruturarMensagens = {
    inicio: `\n${ICONES_COMANDO.reestruturar} Iniciando processo de reestruturação...\n`,
    spinnerCalculandoPlano: `${ICONES_DIAGNOSTICO.progresso} Calculando plano de reestruturação...`,
    planoSugeridoFast: (origem, moverLen) => `${ICONES_STATUS.ok} Plano sugerido (${origem}) FAST: ${moverLen} movimentação(ões)`,
    dryRunFast: 'Dry-run solicitado (--somente-plano). (FAST MODE)',
    reestruturacaoConcluidaFast: (moverLen) => `${ICONES_STATUS.ok} Reestruturação concluída: ${moverLen} movimentos. (FAST MODE)`,
    planoCalculadoFastSemAplicar: 'Plano calculado em FAST MODE (nenhuma ação aplicada sem --auto).',
    erroDuranteReestruturacao: (erroMensagem) => `${ICONES_STATUS.falha} Erro durante a reestruturação: ${erroMensagem}`,
    spinnerPlanoVazio: 'Plano vazio: nenhuma movimentação sugerida.',
    spinnerPlanoSugerido: (origem, moverLen) => `Plano sugerido (${origem}): ${moverLen} movimentação(ões)`,
    spinnerConflitosDetectados: (qtd) => `Conflitos detectados: ${qtd}`,
    spinnerSemPlanoSugestao: 'Sem planoSugestao (nenhum candidato ou erro). Usando ocorrências.',
    dryRunCompleto: 'Dry-run solicitado (--somente-plano). Nenhuma ação aplicada.',
    dicaParaAplicar: 'Para aplicar as movimentações reais, execute novamente com a flag --auto (ou --aplicar).',
    fallbackProblemasEstruturais: (qtd) => `\n${qtd} problemas estruturais detectados para correção:`,
    fallbackLinhaOcorrencia: (tipo, rel, mensagem) => `- [${tipo}] ${rel}: ${mensagem}`,
    nenhumNecessario: 'Nenhuma correção estrutural necessária. Repositório já otimizado!',
    canceladoErroPrompt: `${ICONES_STATUS.falha} Reestruturação cancelada. (Erro no prompt)`,
    canceladoUseAuto: `${ICONES_STATUS.falha} Reestruturação cancelada. (Use --auto para aplicar sem prompt)`,
    spinnerAplicando: `${ICONES_ACAO.correcao} Aplicando movimentos...`,
    reestruturacaoConcluida: (qtd, frase) => `Reestruturação concluída: ${qtd} ${frase}.`,
    falhaReestruturacao: 'Falha na reestruturação.'
};
