// sensei/dist/core/registry/paths.js
import path from 'node:path';
export const PROJETO_RAIZ = process.cwd();
export const SENSEI_DIRS = {
    STATE: path.join(PROJETO_RAIZ, '.sensei'),
    METRICS_HISTORY: path.join(PROJETO_RAIZ, '.sensei', 'historico-metricas'),
    REPORTS: path.join(PROJETO_RAIZ, 'relatorios'),
    PERF: path.join(PROJETO_RAIZ, 'docs', 'perf')
};
export const SENSEI_ARQUIVOS = {
    CONFIG: path.join(PROJETO_RAIZ, 'sensei.config.json'),
    CONFIG_SAFE: path.join(PROJETO_RAIZ, 'sensei.config.safe.json'),
    GUARDIAN_BASELINE: path.join(SENSEI_DIRS.STATE, 'guardian.baseline.json'),
    ESTRUTURA_BASELINE: path.join(SENSEI_DIRS.STATE, 'estrutura.baseline.json'),
    ESTRUTURA_ARQUETIPO: path.join(SENSEI_DIRS.STATE, 'estrutura.arquetipo.json'),
    MAPA_REVERSAO: path.join(SENSEI_DIRS.STATE, 'mapa-reversao.json'),
    REGISTRO_VIGIA: path.join(SENSEI_DIRS.STATE, 'integridade.json'),
    METRICAS_HISTORICO: path.join(SENSEI_DIRS.METRICS_HISTORY, 'metricas-historico.json'),
    GUARDIAN_BASELINE_LEGACY: path.join(SENSEI_DIRS.STATE, 'baseline.json'),
    ESTRUTURA_BASELINE_LEGACY: path.join(SENSEI_DIRS.STATE, 'baseline-estrutura.json'),
    ESTRUTURA_ARQUETIPO_LEGACY_ROOT: path.join(PROJETO_RAIZ, 'sensei.repo.arquetipo.json')
};
export const REPORT_PADROES = {
    DIAGNOSTICO: (timestamp) => path.join(SENSEI_DIRS.REPORTS, `sensei-diagnostico-${timestamp}.md`),
    SUMMARY_JSON: (timestamp) => path.join(SENSEI_DIRS.REPORTS, `sensei-relatorio-summary-${timestamp}.json`),
    ASYNC_ANALYSIS: path.join(SENSEI_DIRS.REPORTS, 'async-analysis-report.json'),
    PERF_BASELINE: (timestamp) => path.join(SENSEI_DIRS.PERF, `baseline-${timestamp}.json`),
    PERF_DIFF: path.join(SENSEI_DIRS.PERF, 'ultimo-diff.json')
};
export const MIGRACAO_MAPA = {
    [SENSEI_ARQUIVOS.GUARDIAN_BASELINE_LEGACY]: SENSEI_ARQUIVOS.GUARDIAN_BASELINE,
    [SENSEI_ARQUIVOS.ESTRUTURA_BASELINE_LEGACY]: SENSEI_ARQUIVOS.ESTRUTURA_BASELINE,
    [SENSEI_ARQUIVOS.ESTRUTURA_ARQUETIPO_LEGACY_ROOT]: SENSEI_ARQUIVOS.ESTRUTURA_ARQUETIPO
};
export function resolveFilePath(newPath) {
    const legacyCaminho = Object.entries(MIGRACAO_MAPA).find(([_, target]) => target === newPath)?.[0];
    return legacyCaminho || newPath;
}
